service/storage_service.cc

```diff
@@ -715,6 +715,7 @@ future<> storage_service::topology_state_load() {
     for (const auto& gen_id : _topology_state_machine._topology.committed_cdc_generations) {
         co_await _cdc_gens.local().handle_cdc_generation(gen_id);
         if (gen_id == _topology_state_machine._topology.committed_cdc_generations.back()) {
+            co_await _sys_ks.local().update_cdc_generation_id(gen_id);
```

Comment:
Are we certain that all nodes manage to save the newly committed generation to `system.cdc_local` before restarting? IIUC, we are, because it happens while applying the group0 command. If it failed on some node, the topology coordinator wouldn't pass the `commit_cdc_generation` transition state.